weldx/core.py

```diff
@@ -540,3 +544,176 @@ def units(self) -> str:
         if isinstance(self._data, xr.DataArray):
             return self._data.data.units
         return self._units
+
+
+# ExternalFile -------------------------------------------------------------------------
+
+
+@dataclass
+class ExternalFile:
+    """Handles the asdf serialization of external files."""
+
+    path: Union[str, Path] = None
+
+    filename: str = None
+    suffix: str = None
+    directory: str = None
+    hostname: str = None
+
+    created: pd.Timestamp = None
+    modified: pd.Timestamp = None
+    size: int = None
+
+    hashing_algorithm: str = "SHA-256"
+    hash: str = None
+    asdf_save_content: bool = False
+    buffer: bytes = None
+    file_system = None
+
+    def __post_init__(self):
+        """Initialize the internal values."""
+        if self.file_system is not None:  # pragma: no cover
+            raise Exception(
+                "Using file systems to load a file is currently not supported."
+            )
```

Comment:
I think we can remove the `file_system` attribute from the class?
I like the way it gets passed to `write_to` when needed 👍

Reply:
We can do it but I thought we should keep it as a reminder since I think it is more useful for actually reading a file (archives, servers, etc) than for writing (Jupyter examples).  --- However you like

Reply:
My vote goes to removing for now and pick this up again if we need it :)